codes/9465_cpp_yji0828.cpp

```diff
@@ -0,0 +1,39 @@
+#define _CRT_SECURE_NO_WARNINGS
+#include <cstdio>
+#include <algorithm>
+#include <iostream>
+using namespace std;
+
+int sticker[2][100001];
+int DP[2][100001];
```

Comment:
arr[2][5]와 arr[5][2]는 얼핏보면 같아보이지만 실제로는 다릅니다.
arr[2][5]는 메모리 상에서 arr[0][0], arr[0][1], ..., arr[0][4], arr[1][0], arr[1][1], ...순이고
arr[5][2]는 arr[0][0], arr[0][1], arr[1][0], ... 순입니다.

아래 코드를 보니 dp[0][n]이 호출될때 dp[1][n]이 같이 호출됩니다. 이때 dp의 선언이 dp[2][100001]이므로 dp[0][n]과 dp[1][n]이 100000 * 4바이트 만큼 떨어져있습니다. 이러면 캐시에서 불리하게 됩니다. 자주 참조하는 메모리끼리 인접하게 설계하는 편이 좋습니다.
```suggestion
int dp[100001][2];
```
그리고 보통 대문자는 변수명이 아닌 class와 같은 곳에서 쓰는데....여기서는 크게 상관없을 것 같지만 일단 코멘트 드립니다.

Reply:
수정했습니다!✔